loadtest/control/simulcontroller/actions.go

```diff
@@ -1840,6 +1840,91 @@ func (c *SimulController) createPostReminder(u user.User) control.UserActionResp
 	return control.UserActionResponse{Info: fmt.Sprintf("created post reminder, id %s", post.Id)}
 }
 
+func (c *SimulController) reconnectWebSocket(u user.User) control.UserActionResponse {
+	team, err := c.user.Store().CurrentTeam()
+	if err != nil {
+		return control.UserActionResponse{Err: control.NewUserError(err)}
+	}
+	if team != nil {
+		// The webapp makes these requests only if the team
+		// is set in the state.
+
+		if _, err := u.GetChannelsForTeamForUser(team.Id, u.Store().Id(), false); err != nil {
+			return control.UserActionResponse{Err: control.NewUserError(err)}
+		}
+
+		if err := u.GetChannelMembersForUser(u.Store().Id(), team.Id); err != nil {
+			return control.UserActionResponse{Err: control.NewUserError(err)}
+		}
+
+		// getPostsForChannel(?since=mostRecentPost.CreateAt in CurrentChannel)
+		crtEnabled, resp := control.CollapsedThreadsEnabled(u)
+		if resp.Err != nil {
+			return resp
+		}
+		currentCh, err := u.Store().CurrentChannel()
+		if errors.Is(err, memstore.ErrChannelNotFound) {
+			return control.UserActionResponse{Info: "current channel is not set"}
+		} else if err != nil {
+			return control.UserActionResponse{Err: control.NewUserError(err)}
+		}
+		posts, err := u.Store().ChannelPostsSorted(currentCh.Id, true)
+		if err != nil {
+			return control.UserActionResponse{Err: control.NewUserError(err)}
+		}
+		latestCreateAt := posts[len(posts)-1].CreateAt
+		if _, err := u.GetPostsSince(currentCh.Id, latestCreateAt, crtEnabled); err != nil {
+			return control.UserActionResponse{Err: control.NewUserError(err)}
+		}
+
+		// unique user ids from all posts in current Channel + users from DM prefs
+		userIds := make(map[string]bool)
+		for _, p := range posts {
+			userIds[p.UserId] = true
+		}
```

Comment:
Could become a `UsersForPosts` util of some kind.

Reply:
done